### PR TITLE
GH1773: Fix NuGetHasSource ArgumentCustomization & add ConfigFile to NuGetSourcesSettings

### LINKED ISSUE #1773: NuGetHasSource call do not take care of ArgumentCustomization in NuGetSourcesSettings
### What You Are Seeing?
When I am using NuGetHasSource with NuGetSourcesSettings including ArgumentCustomization like in the example below the ArgumentCustomization will not applied to the call of the nuget.exe.

```
var nugetSourceSettings = new NuGetSourcesSettings
{
	UserName = "VssSessionToken",
	Password = accessToken,
	IsSensitiveSource = true,
	Verbosity = NuGetVerbosity.Detailed,
	ArgumentCustomization = args => args.Append("-ConfigFile " + solutionDirectory + "/NuGet.config")
};

if (!NuGetHasSource(internalFeed, nugetSourceSettings))
{
	Information("Add internal feed '{0}' as nuget source", internalFeed);
	NuGetAddSource("Internal", internalFeed , nugetSourceSettings);
}
```

**Console Output:**
> 
Executing: "C:/Daimler/FB/fb-timemgmt-service/Build/tools/NuGet.exe" sources List
Registered Sources:

  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
  2.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
  3.  Microsoft Azure Service Fabric SDK [Enabled]
      C:\Program Files\Microsoft SDKs\Service Fabric\packages

### What is Expected?
What I expected is that the nuget.exe will be called like this:

> 
Executing: "C:/Daimler/FB/fb-timemgmt-service/Build/tools/NuGet.exe" sources List -ConfigFile C:/[Path to Repository]/[solutionDirectory]/NuGet.config
Registered Sources:
...

I already checked the implementation of [NuGetSources class](https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/NuGet/Sources/NuGetSources.cs) and it seems the Run function call (L141+142) in the method 'HasSource' will be called with no valid ProcessArgumentBuilder (is null).

### What version of Cake are you using?
0.21.1
### Are you running on a 32 or 64 bit system?
64-Bit
### What environment are you running on?  Windows? Linux? Mac?
Windows
### Are you running on a CI Server?  If so, which one?
VSTS
### How Did You Get This To Happen? (Steps to Reproduce)
See above
### Output Log
No log

### PR DESCRIPTION
**Pull request by devlead**
* Addresses issue with ArgumentCustomization not being used with NuGetHasSource 
* Adds ConfigFile to NuGetSourcesSettings so one can work against custom NuGet.config
* Adds NuGetSourcesSettings  ConfigFile unit tests
* Adds ArgumentCustomization unit test for NuGetHasSource 
* Adds NuGet sources integration tests (_.NET Core on posix excluded as nuget.exe won't run ootb there_)
* Adds NonInteractive which was missing from NuGetHasSource (user input no good in build scripts 😎 )
* Fixes #1773

### CODE REVIEW

File: src/Cake.Common/Tools/NuGet/Sources/NuGetSourcesSettings.cs [hunk 1]
```diff
@@ -44,5 +45,10 @@ public sealed class NuGetSourcesSettings : ToolSettings
         /// <c>true</c> if password is stored as unencrypted; otherwise, <c>false</c>.
         /// </value>
         public bool StorePasswordInClearText { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value the NuGet configuration file. If not specified, file %AppData%\NuGet\NuGet.config is used as configuration file.
```
Comment (gep13) [inline src/Cake.Common/Tools/NuGet/Sources/NuGetSourcesSettings.cs]: @devlead that wording here is a little off.  Perhaps `Gets or sets the location of the NuGet configuration file`

Comment (devlead) [reply src/Cake.Common/Tools/NuGet/Sources/NuGetSourcesSettings.cs]: agreed I'll fix.

Comment (devlead) [reply src/Cake.Common/Tools/NuGet/Sources/NuGetSourcesSettings.cs]: this should be addressed now.

**Comment by gep13** [comment]
@devlead your changes have been merged, thanks for your contribution 👍

**Pull request merged**